libsel4/include/interfaces/object-api.xml

```diff
@@ -825,6 +825,24 @@
             </error>
          </method>
 
+        <method id="TCBSetFlags" name="SetFlags" manual_name="Set Feature Flags" manual_label="tcb_setflags">
```

Comment:
Should this generic invocation support setting and clearing multiple flags at a time? I think it should only allow updating one flag for now. I predict that some flags will require additional capabilities to be set such as flags that control access to other system registers such as genericTimer ctl registers or perf registers. I think it's possible to support a generic invocation that takes flag-specific capability arguments, but not when multiple flag changes are supported in a single call.

Reply:
An additional thing my comment didn't consider is that if you grant authority for a TCB to access a protected system register (like timer or perf registers), then maybe the TCB needs to hold a copy of that capability in a CSlot in the TCB so that the authority can be revoked.

Reply:
I would make that kind of change when we have a concrete proposal for a sys-call that needs it. If it needs a non-TCB cap, that could still be encoded as optional argument(s) that are checked when the corresponding flag is in the set, or it can be excluded from here and be set/unset in a separate syscall.  I don't see any real benefit in removing the generality at this point if we can support it. You can always use the set API to set single flags, but not the other way around.

Reply:
If capabilities are involved, you may not need any flags.

That said, I don't foresee any problems setting or clearing multiple flags, as we only support 4 extra caps and doing 4 cap type lookups in one calls isn't excessive.

And like Gerwin said, we can add such restrictions later if necessary. I'd prefer it to be a requirement of specific flags instead of a generic requirement.